probe-rs/src/flashing/error.rs

```diff
@@ -78,7 +81,29 @@ pub enum FlashError {
     },
     /// An error occurred during the interaction with the core.
     #[error("Something during the interaction with the core went wrong")]
-    Core(#[from] error::Error),
+    Core(#[source] error::Error),
+    /// Failed to reset, and then halt the CPU.
+    #[error("Failed to reset, and then halt the CPU.")]
+    ResetAndHalt(#[source] error::Error),
+    /// Failed to start running code on the CPU.
+    #[error("Failed to start running code on the CPU")]
+    Run(#[source] error::Error),
+    /// Failed to write a CPU register.
+    #[error("Failed to write CPU register {register}.")]
+    WriteRegister {
```

Comment:
Maybe this would make more sense in the general `probe_rs::Error`? These don't seem flash specific, and it would be nice if other users could benefit as well.